### ISSUE TITLE
Libossia Distribution package

### ISSUE THREAD

**Issue by reno-** [OSSIA/libossia#59]
What about distribution of libossia releases?
It will be nice to have an installer for libossia
### OSX
’brew install libossia --with-python’
’brew install libossia --with-python3’
’brew install libossia --with-qt5’
’brew install libossia --with-qml’

### Linux
’apt-get install libossia’

**Comment by reno-** [comment]
If I list cmake options, I understand that there is many bindings.
The distribution is different for each implementations.
Here is the list I've found.

    OSSIA_C    
    OSSIA_FRAMEWORK    
    OSSIA_JAVA    
    OSSIA_LTO    
    OSSIA_MAX    
    OSSIA_PD    
    OSSIA_PYTHON    
    OSSIA_QT    

Each implementation need specific integration in order to deploy latest binaries for end user.

I'm thinking about a `pip install pyossia` to install OSSIA_PYTHON. But we currently have all these implementations in a same repo. CI settings might be a bit complicating for supporting all these things.
What about a repo for each implementation, with OSSIA_C (and/or OSSIA_FRAMEWORK) as submodule for each repo?

**Comment by avilleret** [comment]
iirc @jcelerier prefers to not have one submodule per implementation (at least this was his answer when I asked before starting ossia-pd), this is much easier to maintain (no need update submodule all the time)

concerning the deployment, I have a deken deployment script on my todo list for ossia-pd, and I think this should happen only on tagged release 

and I don't think that CI script will be less complicated with submodule

**Comment by jcelerier** [comment]
I think that we are coming close to at least a beta release in terms of package. I'll try to make them during this week. @reno- @theod @avilleret maybe we could define the set of "target platforms" ? 

For libossia (the C++ part) I'd say 
* Windows (32 / 64 bit)
* macOS (64 bit)
* Ubuntu 16.04 (64 bit)
* Android / iOS ? 

For the extensions, the supported versions should be defined : 
* Supported Pd version (0.47.1 I guess ?)
* Supported Python versions ? 2.7 and 3.6 maybe ? Or even better, make an official python package (in pip) as @reno- mentioned ? 
* I'd say Qt 5.9 which is going to come out in the next few days for Ossia-Qt and will be supported as a LTS version.
* ofxOssia on Windows / Mac / Linux too ? Maybe Android / iOS ?

**Comment by avilleret** [comment]
currently ossia-pd works on cutting edge version of Pd
due to post 0.47-1 API changes, I have to adapt CicmWrapper and ossia-pd to work with last version.
And to be honest it's tedious to support previous version of Pd.
So this beta will be for people who build pd themselves.
Also there is currently no help files and no install script (so it might be boring to make a package ready to use).

**Comment by jcelerier** [comment]
no problemo :p

**Comment by reno-** [comment]
About python, the pyossia module supports at least 2.7 and 3.6 versions of python.
I think we could support from 3.4, but I have to do some tests to be sure.

I started CI jobs for integrations, and I can push to pipi of course.
https://travis-ci.org/PixelStereo/pyossia/jobs/222866825 <https://travis-ci.org/PixelStereo/pyossia/jobs/222866825>

But it requires a bit of work to finish a proper release.

cheers,

renaud

**Comment by jcelerier** [comment]
fantastic, thanks :)

**Comment by reno-** [comment]
I've tried to create a Homebrew Formula, but I don't know why, .so file is copied, but python import does not work.
https://gist.github.com/reno-/072d6ce3e09569aed6d2b78509f1508a

**Comment by reno-** [comment]
Brew Formula is confortable to install libossia for QML, PYTHON, C++ etc…
I re-try now to make a pypi package that build the .so file

Maybe we could have boths ways to do

**Comment by jcelerier** [comment]
hmmm ... maybe this problem ? https://stackoverflow.com/questions/25276329/cant-load-python-modules-installed-via-pip-from-site-packages-directory

**Comment by reno-** [comment]
Tx @jcelerier . I'll have a look
